core/src/main/scala/cats/sequence/sequence.scala

```diff
@@ -17,85 +17,157 @@ trait Sequencer[L <: HList] extends Serializable {
   type F[_]
   type LOut <: HList
   type Out = F[LOut]
+
   def apply(hl: L): Out
 }
 
-private[sequence] abstract class MkHConsSequencer {
+trait MkHNilSequencerLowPrio {
```

Comment:
This is to ensure we can evolve it later while preserving binary compatibility:
```suggestion
private[sequence] abstract class MkHNilSequencerLowPrio {
```